### PR TITLE
Implement Azure Application Gateway create template deployment

### PR DESCRIPTION
**Pull request by milismsft**
Implement Azure Application Gateway create template deployment (couple test cases included as well).

An Application Gateway resource provides an Azure-managed HTTP load-balancing solution based on layer-7 load balancing. These fit better in the Spinnaker load balancer model than the Azure Load Balancer resource that is currently used because it better suits transitioning from one active version of the server group to another. For more info about Azure application gateways and load balancer and how they compare see https://azure.microsoft.com/en-us/documentation/articles/application-gateway-introduction/

### CODE REVIEW

**Comment by milismsft** [comment]
@spinnaker/netflix-reviewers @duftler PTAL

**Comment by milismsft** [comment]
ping

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/view/AzureLoadBalancerProvider.groovy [hunk 1]
```diff
@@ -78,6 +78,7 @@ class AzureLoadBalancerProvider implements LoadBalancerProvider<AzureLoadBalance
       name: loadBalancerDescription.loadBalancerName,
       region: loadBalancerDescription.region,
       vnet: loadBalancerDescription.vnet?: "vnet-unassigned",
+      subnet: loadBalancerDescription.subnet?: "subnet-unassigned",
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/view/AzureLoadBalancerProvider.groovy]: Might read easier with a space before the elvis operator. Can always change this in a subsequent PR.

**Comment by duftler** [comment]
LGTM.
(All changes are contained to azure packages.) 
Will merge when travis completes.

**Pull request merged**